### PR TITLE
vim-patch:partial:cfa8f9a3f285

### LINKED ISSUE #18754: Is section 5 of the registers help page correct?
### Neovim version (nvim -v)

0.7.0

### Vim (not Nvim) behaves the same?

I don't know

### Operating system/version

macOS 12.1

### Terminal name/version

iterm

### $TERM environment variable

does not matter 

### Installation

brew

### How to reproduce the issue

```
:h registers
```
Then go to  section 5

### Expected behavior

When you go to section 5 of the registers documentation, the header mentions three registers `": ". "%` but then the first line refers them with a different name for each one? Or I don't know what that second line is referring to.
I think the first line should contain the same values as the header? Please take a look at the actual behaviour section to see the current content

### Actual behavior

This is what the section says

> 5. Read-only registers ":, ". and "%
> These are '%', '#', ':' and '.'.  You can use them only with the "p", "P",
> and ":put" commands and with CTRL-R.

### PR DESCRIPTION
**Pull request by clason**
Fixes #18754

Update runtime files
https://github.com/vim/vim/commit/cfa8f9a3f285060152ebbdbf86fbc7aecf1dd756

omit:
* translations
* manual rewrite
* most doc changes (for now, @seandewar or @zeertzjq let me know if there's a specific change that is valid for Neovim, then I'll add it)

### CODE REVIEW

File: runtime/doc/map.txt [hunk 1]
```diff
@@ -152,6 +151,22 @@ that mapping won't get expanded yet, Vim is waiting for another character.
...
+separator "|".  Other examples with trailing white space: >
+	unmap @@ 
+	unmap @@     " legacy comment
+An error will be issued, which is very hard to identify, because the ending
+whitespace character in `unmap @@ ` is not visible.
+
+A generic solution is to put the command separator "|" right after the mapped
+keys.  After that white space and a comment may follow: >
+	unmap @@|    " legacy scriptcomment
+
```
Comment (zeertzjq) [inline runtime/doc/map.txt:169]: Is the removal of blank lines intentional?

File: runtime/doc/undo.txt [hunk 2]
```diff
@@ -108,12 +108,14 @@ change again.  But you can do something like this: >
...
 change.
-
+							*undo-break*
 To do the opposite, break a change into two undo blocks, in Insert mode use
 CTRL-G u.  This is useful if you want an insert command to be undoable in
 parts.  E.g., for each sentence.  |i_CTRL-G_u|
+
 Setting the value of 'undolevels' also breaks undo.  Even when the new value
-is equal to the old value.
+is equal to the old value:: >
```
Comment (zeertzjq) [inline runtime/doc/undo.txt:117]:
```suggestion
is equal to the old value: >
```

File: runtime/syntax/vim.vim [hunk 3]
```diff
@@ -367,7 +367,7 @@ syn match	vimSetMod	contained	"&vim\=\|[!&?<]\|all&"
 " Let: {{{2
 " ===
 syn keyword	vimLet	let	unl[et]	skipwhite nextgroup=vimVar,vimFuncVar,vimLetHereDoc
-VimFoldh syn region vimLetHereDoc	matchgroup=vimLetHereDocStart start='=<<\s\+\%(trim\>\)\=\s*\z(\L\S*\)'	matchgroup=vimLetHereDocStop end='^\s*\z1\s*$'
+VimFoldh syn region vimLetHereDoc	matchgroup=vimLetHereDocStart start='=<<\s\+\%(trim\|eval\>\)\=\s*\z(\L\S*\)'	matchgroup=vimLetHereDocStop end='^\s*\z1\s*$'
```
Comment (zeertzjq) [inline runtime/syntax/vim.vim:370]: Needs patch 8.2.4770

**Review by zeertzjq — COMMENTED** [review]
Missing change.txt

File: runtime/doc/map.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/map.txt:169]: No, for some reason they did not get copied from the Github diffview.

**Review by zeertzjq — APPROVED** [review]
Although syntax.txt change itself is applicable, there are too many previous patches that cannot be ported yet.

**Comment by clason** [comment]
> Although syntax.txt change itself is applicable, there are too many previous patches that cannot be ported yet.

Yes, that whole help section is still missing.

**Pull request merged**